src/Filters/RelatedDto.php

```diff
@@ -83,56 +67,146 @@ public function isResolved(string $relationship): bool
         return array_key_exists($relationship, $this->resolvedRelationships);
     }
 
-    public function sync(RestifyRequest $request): self
+    public function reset(): self
     {
-        if (empty($query = ($request->input('related') ?? $request->input('include')))) {
-            $this->loaded = true;
+        $this->loaded = false;
 
-            return $this;
-        }
+        $this->related = RelatedQueryCollection::make([]);
+
+        $this->resolvedRelationships = [];
+
+        $this->relatedArray = [];
 
-        if (! $this->loaded) {
-            $this->related = collect(str_getcsv($query))->mapInto(Stringable::class)->map->ltrim()->map->rtrim()->all();
+        return $this;
+    }
 
-            $this->normalize();
+    private function searchInRelatedQuery(RelatedQuery $relatedQuery, string $relation): ?RelatedQuery
+    {
+        if ($relatedQuery->matchTree($relation)) {
+            return $relatedQuery;
+        }
 
-            $this->loaded = true;
+        if ($relatedQuery->nested->count()) {
+            return $relatedQuery->nested->first(fn (RelatedQuery $child) => $this->searchInRelatedQuery(
+                $child,
+                $relation
+            ));
         }
 
-        return $this;
+        return null;
     }
 
-    public function reset(): self
+    private function makeTreeForChild(RelatedQuery $relatedQuery, array &$base = []): array
     {
-        $this->loaded = false;
+        if ($relatedQuery->nested->count()) {
+            $relatedQuery->nested->each(function (RelatedQuery $child, $i) use (&$base, $relatedQuery) {
+                $base[$i] = data_get($base, $i, $relatedQuery->relation).".$child->relation";
+
+                if ($child->nested->count()) {
+                    $this->makeTreeForChild($child, $base);
+                }
+            });
+        } else {
+            return [$relatedQuery->relation];
```

Comment:
Return this in `if (! $related->nested->count()` or `if (...->count == 0)` and loose an indentation level for the big piece of code